jax/lax/lax_control_flow.py

```diff
@@ -543,7 +543,68 @@ def _while_transpose_error(*_, **kwargs):
 batching.primitive_batchers[while_p] = _while_loop_batching_rule
 
 
-### cond
+### cond and switch
+
+def switch(index, branches: Sequence[Callable], operand):
+  """Apply exactly one of ``branches`` given by ``index``.
+
+  If ``index`` is out of bounds, it is clamped to within bounds.
+
+  Has the semantics of the following Python::
+
+    def switch(index, branches, operand):
+      index = clamp(0, index, len(branches) - 1)
+      return branches[index](operand)
+
+  Arguments:
+    index: Integer scalar type, indicating which branch function to apply.
+    branches: Sequence of functions (A -> B) to be applied based on `index`.
+    operand: Operand (A) input to whichever branch is applied.
+  """
+  if len(onp.shape(index)) != 0:
+    raise TypeError(
+        f"Branch index must be scalar, "
+        f"got {index} of shape {onp.shape(index)}.")
+
+  try:
+    index_dtype = dtypes.result_type(index)
+  except TypeError as err:
+    msg = f"Index type must be an integer, got {index}."
+    raise TypeError(msg) from err
+
+  if index_dtype.kind not in 'iu':
+    raise TypeError(
+        f"Index type must be an integer, got {index} as {index_dtype}")
+
+  branches = tuple(branches)
+
+  if len(branches) == 0:
+    raise ValueError("Empty branch sequence")
+  elif len(branches) == 1:
+    return branches[0](operand)
+
+  index = lax.clamp(0, index, len(branches) - 1)
+
+  if (jax.api._jit_is_disabled() and
+      isinstance(core.get_aval(index), ConcreteArray)):
+    return branches[index](operand)
```

Comment:
```suggestion
    return branches[int(index)](operand)
```

I think if `index` were a Tracer with a concrete aval, `branches[index]` could still fail because (1) lists don't like to be indexed by our Tracer objects and (2) what if it's of float or complex dtype.

(I tend to like to test `int(index)` and catch the exception, rather than the isinstance check, but only because it depends on fewer api details. Not a big deal so use whatever you perfer!)

Reply:
Good catch! Done.